app/Http/Controllers/api/v1/auth/RegisterController.php

```diff
@@ -3,63 +3,70 @@
 namespace App\Http\Controllers\api\v1\auth;
 
 use App\Http\Controllers\Controller;
+use App\Http\Requests\StoreUser;
+use App\Http\Requests\StoreUserInvitation;
+use App\Invitation;
 use App\User;
-use Illuminate\Foundation\Auth\RegistersUsers;
+use Illuminate\Http\JsonResponse;
 use Illuminate\Support\Facades\Hash;
-use Illuminate\Support\Facades\Validator;
+use Illuminate\Support\Facades\Lang;
 
 class RegisterController extends Controller
 {
-    /*
-    |--------------------------------------------------------------------------
-    | Register Controller
-    |--------------------------------------------------------------------------
-    |
-    | This controller handles the registration of new users as well as their
-    | validation and creation. By default this controller uses a trait to
-    | provide this functionality without requiring any additional code.
-    |
-    */
-
-    use RegistersUsers;
-
     /**
-     * Create a new controller instance.
-     *
-     * @return void
+     * Create a new user
+     * @param  StoreUser    $request
+     * @return JsonResponse
      */
-    public function __construct()
+    public function register(StoreUser $request)
     {
-        $this->middleware('guest');
-    }
+        $user = new User();
 
-    /**
-     * Get a validator for an incoming registration request.
-     *
-     * @param  array                                      $data
-     * @return \Illuminate\Contracts\Validation\Validator
-     */
-    protected function validator(array $data)
-    {
-        return Validator::make($data, [
-            'name'     => ['required', 'string', 'max:255'],
-            'email'    => ['required', 'string', 'email', 'max:255', 'unique:users'],
-            'password' => ['required', 'string', 'min:8', 'confirmed'],
-        ]);
+        $user->firstname = $request->firstname;
+        $user->lastname  = $request->lastname;
+        $user->username  = $request->username;
+        $user->email     = $request->email;
+        $user->password  = Hash::make($request->password);
+
+        $store = $user->save();
+
+        return ($store === true) ? (response()->json($user, 201)) : (response()->json(['message' => Lang::get('validation.custom.request.400')], 400));
     }
 
     /**
-     * Create a new user instance after a valid registration.
-     *
-     * @param  array     $data
-     * @return \App\User
+     * Create a new user through invitation register method
+     * @param  StoreUserInvitation $request
+     * @return JsonResponse
      */
-    protected function create(array $data)
+    public function invitationRegister(StoreUserInvitation $request)
```

Comment:
Please merge both methods together

Reply:
https://github.com/THM-Health/PILOS/pull/23/commits/1a33a7aa89dddde61afd7c0994421ca8d1e20bfb merged to a single method